### PR TITLE
Remove dependency on EditorFeatures from Remote.ServiceHub project - take 2

### PR DESCRIPTION
**Pull request by tmat**
Redo of https://github.com/dotnet/roslyn/pull/59059.

Requires Type Script change: https://devdiv.visualstudio.com/DevDiv/_git/TypeScript-VS/pullrequest/376624

### CODE REVIEW

File: src/Features/Core/Portable/Microsoft.CodeAnalysis.Features.csproj [hunk 1]
```diff
@@ -128,10 +129,12 @@
     <PublicAPI Include="PublicAPI.Unshipped.txt" />
   </ItemGroup>
   <ItemGroup>
+    <PackageReference Include="System.Text.Json" Version="$(SystemTextJsonVersion)" />
     <PackageReference Include="System.Threading.Tasks.Extensions" Version="$(SystemThreadingTasksExtensionsVersion)" />
     <PackageReference Include="Microsoft.DiaSymReader" Version="$(MicrosoftDiaSymReaderVersion)" />
     <PackageReference Include="Microsoft.CodeAnalysis.AnalyzerUtilities" Version="$(MicrosoftCodeAnalysisAnalyzerUtilitiesVersion)" />
-    <PackageReference Include="System.Text.Json" Version="$(SystemTextJsonVersion)" />
+    <PackageReference Include="Microsoft.CodeAnalysis.Elfie" Version="$(MicrosoftCodeAnalysisElfieVersion)" />
+    <PackageReference Include="Microsoft.VisualStudio.RemoteControl" Version="$(MicrosoftVisualStudioRemoteControlVersion)" />
```
Comment (JoeRobich) [inline src/Features/Core/Portable/Microsoft.CodeAnalysis.Features.csproj:137]: Do we have to bring a VS reference into the Features layer?

Comment (JoeRobich) [reply src/Features/Core/Portable/Microsoft.CodeAnalysis.Features.csproj:137]: Also, are these source buildable since Features ships in the SDK?

Comment (tmat) [reply src/Features/Core/Portable/Microsoft.CodeAnalysis.Features.csproj:137]: Hmm, good point. These are not VS dependencies, but source build is a problem.

Comment (tmat) [reply src/Features/Core/Portable/Microsoft.CodeAnalysis.Features.csproj:140]: Fixed.

File: src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngineProxy.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                (service, cancellationToken) => service.FindReferenceAssembliesWithTypeAsync(name, arity, cancellationToken),
+                cancellationToken).ConfigureAwait(false);
+
+            return result.HasValue ? result.Value : ImmutableArray<ReferenceAssemblyWithTypeResult>.Empty;
+        }
+
+        public async ValueTask UpdateContinuouslyAsync(string sourceName, string localSettingsDirectory, ISymbolSearchLogService logService, CancellationToken cancellationToken)
+        {
+            // logService parameter is ignored since it's already set on the connection as a callback
+            _ = logService;
```
Comment (JoeRobich) [inline src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngineProxy.cs:54]: Is this just to avoid `logService` being unused?

**Review by JoeRobich — APPROVED** [review]

File: src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngineProxy.cs (hunk 2 above)
Comment (tmat) [reply src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngineProxy.cs:54]: I guess. Just moved existing code.

**Pull request merged**